add OIDC login

markbeep opened the pull request:
Closes #33

markbeep commented:
Because of some refactoring this PR turned out a lot bigger than I planned. Am not satisfied with the changes at all. It's very smelly, but it should work.

I'll look into cleaning it up some before merging it.

MichelfrancisBustillos commented:
@markbeep I run Authentik for SSO and am happy to beta test this if you'd like!

MichelfrancisBustillos commented:
@markbeep Dumb question, how do I pull the docker image for this commit, or do I need to compile it? I am running the `:nightly` branch currently but don't see OIDC under 'Settings > Security'

markbeep commented:
> @markbeep Dumb question, how do I pull the docker image for this commit, or do I need to compile it? I am running the `:nightly` branch currently but don't see OIDC under 'Settings > Security'

@MichelfrancisBustillos yes you'd have to checkout this branch and run it or the docker image locally. I added a docker-compose file so you can run it by simple executing `docker compose up --build` locally.

The `:nightly` version is for anything that is already in the master-branch

MichelfrancisBustillos commented:
@markbeep Getting "failed to solve: Internal: open /proc/stat: transport endpoint is not connected" error when trying to build. I'll admit my Docker knowledge is not the greatest, so it's likely an issue on my end.

MichelfrancisBustillos commented:
@markbeep I was able to get it up and running but am running into an issue with the Redirect URI. 
I currently have "https://audiobookrequests.katoteros.org/auth/oidc" but it returns and invalid or mismatched URI error.

markbeep commented:
Thanks a lot for the feedback! Helps a lot.

As for your points:

- Correctly handle logging out now and have it redirect to your oidc's logout endpoint.
- Did you add the "email" scope? Authentik requires that scope to be able to use the email claim.
- Added an explanation of the roles in the "users" section.
- Fixed it so only if you have a oidc group assigned will it override instead of always defaulting back to "untrusted"

markbeep commented:
@MichelfrancisBustillos I've decided to go ahead and merge it. It'll be available under the `:nightly` tag now. I'll wait a few days before adding it to a release though.

Healzangels commented:
@MichelfrancisBustillos Hey saw you were setting ODIC up with Authentik. I had been working to achieve the same thing but came up a bit short so far. If you had the time/willing was hoping you might be able to go into more depth on your configuration settings. Cheers!

MichelfrancisBustillos commented:
@Healzangels Feel free to reach out to me via Discord (@the_armed_programmer) or email (michelfrancisb@gmail.com) and I'd be happy to help you out!
